### PR TITLE
#15449 : Sanitize bundle file name when uploading bundle using Micros…

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…oft Edge on Windows 10

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publisher/ajax/RemotePublishAjaxAction.java [hunk 1]
```diff
@@ -713,19 +713,18 @@ public void uploadBundle ( HttpServletRequest request, HttpServletResponse respo
...
+        String bundleName = null;
         try {
+            @SuppressWarnings ("unchecked")
+            List<FileItem> items = (List<FileItem>) upload.parseRequest( request );
+            InputStream bundle = items.get( 0 ).getInputStream();
+            bundleName = sanitizeBundleName(items.get(0).getName());
+            String bundlePath = ConfigUtils.getBundlePath() + File.separator;
+            String bundleFolder = bundleName.substring( 0, bundleName.indexOf( ".tar.gz" ) );
+            String endpointId = getUser().getUserId();
+            response.setContentType( "text/html; charset=utf-8" );
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/ajax/RemotePublishAjaxAction.java]: it wouldn't a bad idea to set this as parameter, eventually someone might need a diff charset

File: dotCMS/src/main/java/com/dotcms/publisher/ajax/RemotePublishAjaxAction.java [hunk 2]
```diff
@@ -738,11 +737,36 @@ public void uploadBundle ( HttpServletRequest request, HttpServletResponse respo
...
+    /**
+     * Checks the name of the bundle file coming from the request to make sure that it is correct. For example, in
+     * Microsoft Edge on Windows 10, the name includes the absolute path to the file instead of the file name only. We
+     * need to strip the absolute path away.
+     *
+     * @param bundleName The name of the bundle file.
+     *
+     * @return The sanitized bundle file name.
+     */
+    private String sanitizeBundleName(String bundleName) throws DotPublisherException {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/publisher/ajax/RemotePublishAjaxAction.java]: @jcastro-dotcms Place somewhere else? Maybe a util? BundleUtil? and a unit test should be nice

Comment (jgambarios) [reply dotCMS/src/main/java/com/dotcms/publisher/ajax/RemotePublishAjaxAction.java]: Done, moved to the `BundleUtil`

Comment (jgambarios) [reply dotCMS/src/main/java/com/dotcms/publisher/ajax/RemotePublishAjaxAction.java]: Also de unit test was created.

**Review by dsilvam — APPROVED** [review]

**Pull request merged**